samples/build-duration-logger/index.js

```diff
@@ -0,0 +1,255 @@
+const EventSourcePolyfill = require('eventsource');
+
+// The address of your Gradle Enterprise server
+const GRADLE_ENTERPRISE_SERVER_URL = process.argv.slice(2);
+const EXPORT_API_USER = process.env.EXPORT_API_USER;
+const EXPORT_API_PASSWORD = process.env.EXPORT_API_PASSWORD;
+const BASIC_AUTH_TOKEN = Buffer.from(`${EXPORT_API_USER}:${EXPORT_API_PASSWORD}`).toString('base64')
+
+// The point in time from which builds should be processed.
+// Values can be 'now', or a number of milliseconds since the UNIX epoch.
+// The time is the point in time that the build was published to the server.
+const PROCESS_FROM = 'now';
+
+// How many builds to process at one time.
+// If running with very fast network connection to the server,
+// this number can be increased for better throughput.
+const MAX_CONCURRENT_BUILDS_TO_PROCESS = 6;
+
+// A build event handler that calculates and logs the build duration.
+//
+// Each "on" method is an event handler that receives each instance of that type of event.
+// Please see https://docs.gradle.com/enterprise/export-api for more information about the event types.
+//
+// After all events have been received, the "complete()" method will be called if it exists.
+class BuildDurationEventsHandler {
+    constructor(build) {
+        this.buildId = build.buildId;
+    }
+
+    onBuildStarted(eventPayload) {
+        this.startTime = eventPayload.timestamp;
+    }
+
+    onBuildFinished(eventPayload) {
+        const endTime = eventPayload.timestamp;
+        console.log(`Build ${this.buildId} completed in ${endTime - this.startTime}ms`);
```

Comment:
Maybe prefix the `buildId` with the Gradle Enterprise server URL such that it's clickable in the terminal.

Reply:
done